src/test/codegen/try_identity.rs

```diff
@@ -7,11 +7,28 @@
 
 type R = Result<u64, i32>;
 
+// This was written to the `?` from `try_trait`,
+// but `try_trait_v2` uses a different structure,
+// so the relevant desugar is copied inline
+// in order to keep the test testing the same thing.
```

Comment:
can this test be rewritten to test the same thing for `TryV2` and if so, does that already optimize to the same thing?

Reply:
From what I've tried, it appears that it does not optimize to the same thing.  My hypothesis is that the mir-opt can only handle Result->Result, but the new desugaring is Result->ControlFlow->Result, so it would need to collapse *both* `match`es to see the identity transform, and I don't think it looks for that.

Filed https://github.com/rust-lang/rust/issues/85133 to track writing a stronger optimization, and mentioned it in a FIXME for this one and the following two.  Perhaps a case-of-case optimization to remove the intermediate `ControlFlow`.  I don't think that blocks checkin, though, since `SimplifyArmIdentity` is currently off by default (because of things like https://github.com/rust-lang/rust/issues/78628).